io/net/bridge.py

```diff
@@ -92,6 +96,9 @@ def test_bridge_delete(self):
         Set to original state
         '''
         self.check_failure('ip link del dev %s' % self.bridge_interface)
+        path = "/etc/sysconfig/network-scripts/ifcfg-%s" \
+               % self.bridge_interface
+        os.remove(path)
```

Comment:
@smruti77 how is this handled for normal interface, in other tests ?
I want same process to be followed for all types of interfaces, by network utils.

Reply:
as for other cases, we are doing this way using 
https://github.com/avocado-framework-tests/avocado-misc-tests/blob/11597cc6cd465827b93e61bb00919ad2c587a952/io/net/Network_config.py#L161

as you see restore_backup used backup interface if bridge always be the case to remove we can be code having check interface type but did not feel any harm to using os.remove, any way @smruti77 can take a call :)

Reply:
@PraveenPenguin We have been doing a lot of work in the past months to push a lot of functionality to utils, and then use that to make common parts of code a lot more uniform. This, maintenance of code in future becomes easier. So, I would prefer adding hooks to `restore_from_backup()` for bridge as well.

@smruti77 your views please

Reply:
@narasimhan-v sure got you, so in bride case, we need to remove always the interface file?

Reply:
@narasimhan-v for all interface why need removed of config file, any way we are restoring the original config.

Reply:
@smruti77 can answer this exactly, since I am unaware of what differences bridges have to normal interfaces, and what hooks are needed.

Reply:
> @narasimhan-v for all interface why need to removed of config file, any way we are restoring the original config from backup.

@smruti77 no his point is we can resue  `restore_from_backup` having bridge support as well 
so my question is in bride case we always remove interface file or we can tweak another way i can quickly come up with lib patch

Reply:
@smruti77 Can this file removal be done by `restore_from_backup()` or `unset_ip()` ?

Reply:
if not, we should add this to one of those functions.

Reply:
@narasimhan-v it will be not done by any of the function, but we don't want to removed config file always.

Reply:
@narasimhan-v for bridge interface actually we are  facing issue, after remove the interface also, if we will call ifup br0 that interfcae is coming back,because of config file. so i deleted that.but for other interface we don't need to remove config file.

Reply:
@narasimhan-v another thing if i will add remove config file here, restore_from_backup() .
then what is use of move backup file.

Reply:
@smruti77 What I wanted to say is: can we add a special case in one of those 2 util functions (wherever applicable), to remove this file for bridge interfaces ?

Do I make sense ?

Reply:
@narasimhan-v let it be like this for now, if we need  for more test then we will add it to util. that will be better.

Reply:
> @narasimhan-v for bridge interface actually we are facing issue, after remove the interface also, if we will call ifup br0 that interfcae is coming back,because of config file. so i deleted that.but for other interface we don't need to remove config file.

just thinking is this not a bug ?

Reply:
> > @narasimhan-v for bridge interface actually we are facing issue, after remove the interface also, if we will call ifup br0 that interfcae is coming back,because of config file. so i deleted that.but for other interface we don't need to remove config file.
> 
> just thinking is this not a bug ?

@smruti77 ^^

Reply:
> @narasimhan-v let it be like this for now, if we need for more test then we will add it to util. that will be better.

Okay, though not ideal case, makes sense.

Reply:
> > > @narasimhan-v for bridge interface actually we are facing issue, after remove the interface also, if we will call ifup br0 that interfcae is coming back,because of config file. so i deleted that.but for other interface we don't need to remove config file.
> > 
> > 
> > just thinking is this not a bug ?
> 
> @smruti77 ^^

@narasimhan-v  @PraveenPenguin i checked in abdul machine and in mine machine also same thing is happening.

Reply:
@narasimhan-v @PraveenPenguin 
1. for a normal interface, we expect the cfg file created and always stays in sysfs  so backup restore works as per utils
2. in case of bridge, the bridge is created using bridge utils and not using lib utils... so when bridge interface is created from test, the distro is basically creating a cfg file for br0 (which is odd and not same on all distros). now there is no need for br0 sysfs entry to stay ... as this will effect other interface as br0 will use same ip of backing interface... 

so at the end of bridge test it is cleaning up the br0 config created on the fly.. but we are not cleaning the br0 sysfs entry created by distro

as per your suggestion, yes it is always good to push all the bridge creation/deletion utils to lib.. we will take this as TODO

@smruti77

Reply:
this patch looks good to me for now.. it can be merge

Reply:
@smruti77 please add a TODO, similar to https://github.com/avocado-framework-tests/avocado-misc-tests/blob/master/io/net/virt-net/lpm.py#L256

Reply:
Once done, we can merge it.

Reply:
ok

Reply:
@abdhaleegit thanks for the ACK. Merging